framework/src/eventloop/eventloop_timer.c

```diff
@@ -59,10 +62,10 @@ static void timeout_callback_func(el_timer_t *timer)
 	}
 }
 
-el_timer_t *eventloop_add_timer(unsigned int timeout, bool repeat, timeout_callback func, void *data)
+static el_timer_t *add_timer(el_loop_t *loop, unsigned int timeout, bool repeat, timeout_callback func, void *data)
 {
-	el_timer_t *timer = NULL;
-	timer_cb_t *callback = NULL;
+	el_timer_t *timer;
+	timer_cb_t *callback;
 
 	timer = (el_timer_t *)EL_ALLOC(sizeof(el_timer_t));
```

Comment:
Why don't you check whether the input parameters are valid before any memory allocation?